File: infra/src/main/java/com/beat/infra/persistence/promotion/mapper/PromotionPersistenceMapper.java
package com.beat.infra.persistence.promotion.mapper;

import org.springframework.stereotype.Component;

import com.beat.domain.promotion.domain.Promotion;
import com.beat.infra.persistence.promotion.entity.PromotionJpaEntity;

@Component
public class PromotionPersistenceMapper {

	public Promotion toDomain(PromotionJpaEntity entity) {
		return Promotion.rehydrate(
			entity.getId(),
			entity.getPromotionPhoto(),
			entity.getPerformanceId(),
			entity.getRedirectUrl(),
			entity.isExternal(),
			entity.getCarouselNumber()
		);
	}

	public PromotionJpaEntity toEntity(Promotion promotion) {
		return PromotionJpaEntity.rehydrate(
			promotion.getId(),
			promotion.getPromotionPhoto(),
			promotion.getPerformanceId(),
			promotion.getRedirectUrl(),
			promotion.isExternal(),
			promotion.getCarouselNumber()
		);
	}
}

File: core/infra/src/main/kotlin/com/beat/infra/persistence/promotion/mapper/PromotionPersistenceMapper.kt
package com.beat.infra.persistence.promotion.mapper

import com.beat.domain.promotion.model.Promotion
import com.beat.infra.persistence.promotion.entity.PromotionJpaEntity
import org.springframework.stereotype.Component

@Component
class PromotionPersistenceMapper {

    fun toDomain(entity: PromotionJpaEntity): Promotion =
        Promotion.rehydrate(
            entity.id,
            entity.promotionPhoto,
            entity.performanceId,
            entity.redirectUrl,
            entity.isExternal,
            entity.carouselNumber,
        )

    fun toEntity(promotion: Promotion): PromotionJpaEntity =
        PromotionJpaEntity.rehydrate(
            promotion.getId(),
            promotion.promotionPhoto,
            promotion.getPerformanceId(),
            promotion.redirectUrl,
            promotion.isExternal,
            promotion.carouselNumber,
        )
}
